rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -393,7 +423,9 @@ impl ByteStream {
     /// # }
     /// ```
     pub fn into_async_read(self) -> impl tokio::io::AsyncRead {
```

Comment:
Earlier we could do this wrapping ourselves & since `tokio_util::io::StreamReader` implements `AsyncBufRead`, we would get a buffered reader for free. Would it be acceptable to change this to:
```suggestion
    pub fn into_async_read(self) -> impl tokio::io::AsyncBufRead {
```

Reply:
Or perhaps you would want a separate `into_async_buf_read`?

Reply:
We don't have any buffering in our implementation. You should be able to wrap the ByteStream with a new-type and implement Stream for it to get the same functionality you used to have. The Stream impl should be a simple call through to the poll_next and size_hint methods.

Reply:
Yes, I know. What I'm saying is that in this function, you're wrapping bytestream in `StreamReader` from tokio, which implements `AsyncBufRead`. So you could also return `impl AsyncBufRead` from this function. 
Otherwise, one has to wrap this in a `BufReader` for no reason, since it is not possible to wrap `ByteStream` in `StreamReader` easily (without wrapping & implementing Stream manually).

Reply:
Oh, I see what you're saying. Yeah, I don't see why we couldn't have two methods. You're welcome to submit a PR to add that.

Reply:
I would say it makes more sense to change the signature of the existing function as I suggested above. It won't be a breaking change at all since AsyncBufRead is a super trait of AsyncRead.

Reply:
Sounds good.